libs/store/src/lib/infrastructure/persistence/domain/base-classes/base-entity.ts

```diff
@@ -0,0 +1,47 @@
+import { Composite } from './composite';
+
+/**
+ * BaseEntityDTO that extends passed Entity DTO
+ * */
+export interface BaseEntityDTO {
+    id: number;
```

Comment:
Are we expecting the primary key of a DTO to always be named `id` and be of type `number`? In many cases the primary key will be named something else and be of type `string`.

I think we allow the user to set what the primary key is in the Entity decorator:
https://github.com/SAP/fundamental-ngx/blob/ab0dad82f415e3fc2150c5f45b507c6c9d01276d/libs/store/src/lib/domain/entity.ts#L11

Reply:
There could be variety of ways to specify which key is primary key. In ariba especially we used many times. This is the reason we started originally with `Identity`  so we can describe what field and type. as kevin says we need to be able to specify which field is primary key